docs/extending-and-experimenting.md

```diff
@@ -147,3 +147,130 @@ subscribe to the result of `executeQuery`, keep track
 of the unsubscription (`teardown`) and update
 some state. This all can be reapplied when you write your
 own APIs.
+
+## Making your own Exchange
+
+As with any library, you're going to find yourself wanting to change (or adapt) the way things work.
+Since day one, with `urql`, we have aimed to make adaptability our number one priority through the use of _Exchanges_.
+Lets run through how this works by creating our own Dedup exchange - a middleware to debounce incoming query operations.
+
+> Note: We've already provided a bunch of exchanges that we thought you would find
+> useful. Check out the [exchanges source code](https://github.com/FormidableLabs/urql/blob/master/src/exchanges) for examples and usage information.
+
+Before you begin, it is assumed that you understand the fundamentals of _Operations_
+and _Exchanges_ in `urql`. There's a detailed description in the [Architecture](./architecture.md) section of the docs, but here's a quick recap:
+
+- Every operation in `urql` is passed to exchanges
+- Each exchange can do one of two things
+  - Forward the operation to the next exchange in line
+  - Return a result of the operation to the client
+
+First thing first, lets set up our boilerplate by making an _Exchange_ that takes an operation stream and passes it immediately to the next exchange in line. We'll also add in some initial logic which will later allow us to track which operations are in progress.
+
+```ts
+import { filter, pipe, tap } from 'wonka';
+import { Exchange } from 'urql';
+
+export const dedupExchange: Exchange = ({ forward }) => {
+  const inFlight = new Set<string>();
+
+  return ops$ => forward(ops$);
+};
+```
+
+You can see from the above example that an exchange is a [curried function](https://medium.com/javascript-scene/curry-and-function-composition-2c208d774983). It takes some setup arguments and returns a function which receives an operation stream (`ops$`) and forwards it to the next exchange (`forward`).
```

Comment:
I'd prefer not to point to Medium. Docs like Redux' one used to explain this fine without and I'm pretty sure I've already got an explanation for this in the Architecture section that avoids complex terms